### ISSUE TITLE
Configure error: execve(./_build/default/tools/configure/configure.exe): Operation not supported on socket

### ISSUE THREAD

**Issue by barracuda156** [coq/coq#17142]
```
:info:configure Executing:  cd "/opt/local/var/macports/build/_opt_PPCRosettaPorts_lang_coq/coq/work/coq-8.16.1" && ./configure -prefix /opt/local -mandir /opt/local/share/man -coqdocdir /opt/local/share/coq/latex -browser 'open %s' -coqide no -with-doc no 
:debug:configure system:  cd "/opt/local/var/macports/build/_opt_PPCRosettaPorts_lang_coq/coq/work/coq-8.16.1" && ./configure -prefix /opt/local -mandir /opt/local/share/man -coqdocdir /opt/local/share/coq/latex -browser 'open %s' -coqide no -with-doc no 
:info:configure Error:
:info:configure execve(./_build/default/tools/configure/configure.exe): Operation not supported on socket
:info:configure Command failed:  cd "/opt/local/var/macports/build/_opt_PPCRosettaPorts_lang_coq/coq/work/coq-8.16.1" && ./configure -prefix /opt/local -mandir /opt/local/share/man -coqdocdir /opt/local/share/coq/latex -browser 'open %s' -coqide no -with-doc no 
:info:configure Exit code: 1
:error:configure Failed to configure coq: configure failure: command execution failed
:debug:configure Error code: NONE
```

Anything can be done about this? It is on old macOS, but we still use and support those :)
```
:debug:main Starting logging for coq @8.16.1_0
:debug:sysinfo Mac OS X 10.6.8 (darwin/10.8.0) arch powerpc
:debug:sysinfo MacPorts 2.8.0
:debug:sysinfo Xcode 3.2.6, CLT 1.0.0.9000000000.1.1248867338
:debug:sysinfo SDK 10.6
:debug:sysinfo MACOSX_DEPLOYMENT_TARGET: 10.6
```
I should add that I have OCaml 4.14.0 and GCC 12.2.0.

P. S. Earlier version of `coq` builds perfectly fine:
```
10:~ svacchanda$ port -v installed coq
The following ports are currently installed:
  coq @8.13.2_0 (active) requested_variants='' platform='darwin 10' archs='ppc' date='2023-01-08T03:32:42+0800'
```

**Comment by ejgallego** [comment]
@barracuda156 can you try building master manually instead of by macports and see how far you reach?

**Comment by SkySkimmer** [comment]
This seems like a dune issue not a coq issue?

**Comment by ejgallego** [comment]
Hard to know without more info. Please reopen with extra info.

**Comment by barracuda156** [comment]
@ejgallego Sorry for a pause, been busy with R ports. I think I have no control to re-open an issue here, but I will comment with update once I get some results.

**Comment by ejgallego** [comment]
No worries at all, if you are still interested in investigating this let's do it!

**Comment by barracuda156** [comment]
> No worries at all, if you are still interested in investigating this let's do it!

@ejgallego Unfortunately, I get the identical failure at configure. Tried with master now, tried with 8.15.2.

Once configure manner was changed after 8.13.x, everything is broken. There is no additional output, since it fails at the very beginning.

**Comment by ejgallego** [comment]
Sorry to hear that @barracuda156 , thanks for the reproduction problem, may be worth checking if there is a problem upstream with Dune https://github.com/ocaml/dune/issues and your OSX version.

What's your Dune version by the way?

Unfortunately due to lack of the right OSX hardware I can't help a lot more debugging.

You may be able to workaround this problem by manually doing:
```
$ dune build ./tools/configure/configure.exe
$ _build/default/tools/configure/configure.exe $options
```
Let me know how that goes.

**Comment by barracuda156** [comment]
@ejgallego I just updated `dune` to 3.7.0 (had some intermediary version between 3.6.0 and 3.7.0), and it did not help.

You suggestion works however, compilation starts. Thank you!

**Comment by ejgallego** [comment]
Glad to hear it helped, it seems your setup is complex, with OSX 10.6 (2009) and Rosetta is involved too. Why is Rosetta involved tho, are you using a PPC Mac? If so, why not compile using the PPC native Ocaml compiler?

But indeed, the problem seems related to `dune exec` not interacting properly with some trick OSX may be doing on binaries, I dunno what to do here, you could report the problem to dune upstream, but given the age of this setup I doubt we could make a lot of progress on it.

**Comment by ejgallego** [comment]
I actually see you tried to revive OCaml PPC, but indeed, for 5.0 you will face a lot of work; that hardware / OS combo is too tricky to support, tho if I can help in some way I'd be happy to.

It think on Coq side the best we can do is to provide the workaround, I will rename this issue, feel free to reopen if you face any more trouble, good luck!

**Comment by barracuda156** [comment]
@ejgallego Thank you.

Why Rosetta: sadly, there is no way to have G5 in a laptop, so if I am out of station, all testing is done in Rosetta on 10.6.8. But right now I am building natively on a PowerMac.

`coq` went for a while and failed on:
```
DUNE      _build/default/coqide-server.install
File "topbin/dune", line 22, characters 7-15:
22 |  (name coqc_bin)
            ^^^^^^^^
Error: No rule found for topbin/coqc_bin.exe
make[1]: *** [_build/default/coq-core.install] Error 1
make[1]: *** Waiting for unfinished jobs....
File "ide/coqide/dune", line 12, characters 7-13:
12 |  (name idetop)
            ^^^^^^
Error: No rule found for ide/coqide/idetop.exe
```
Looks like it ignored configure arg `-coqide no` and build target `world`?
This is 8.16.1.

I use OCaml 4.14.0 at the moment, bytecode compiler (ocamlopt is broken, not hopelessly, but no time to fix it properly).

**Comment by barracuda156** [comment]
@ejgallego Actually configure looks sensible:
```
--->  Configuring coq
Executing:  cd "/opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_lang_coq/coq/work/coq-8.16.1" && _build/default/tools/configure/configure.exe -prefix /opt/local -mandir /opt/local/share/man -docdir /opt/local/share/coq/latex -browser 'open %s' -coqide no -with-doc no 
You have OCaml 4.14.0. Good!
You have OCamlfind 1.9.6. Good!
Warning: Cannot find the OCaml native-code compiler.
Only the bytecode version of Coq will be available.
You have the Zarith library 1.12 installed. Good!
CoqIde manually disabled:
=> no CoqIDE will be built.

  Architecture                : Darwin
  Sys.os_type                 : Unix
  OCaml version               : 4.14.0
  OCaml binaries in           : /opt/local/bin/
  OCaml library in            : /opt/local/lib/ocaml
  CoqIDE                      : no
  Documentation               : None
  Web browser                 : open %s
  Coq web site                : http://coq.inria.fr/
  Bytecode VM enabled         : true
  Native Compiler enabled     : ondemand
```
What may be failing? Or is 8.16.1 not supported with OCaml 4.x?

**Comment by ejgallego** [comment]
@barracuda156 the problem is that you are building in bytecode, for that in 8.16 you need to pass `-byte-only` to configure.

**Comment by ejgallego** [comment]
```
Error: No rule found for topbin/coqc_bin.exe
```
This is indeed because Coq tries to build `coqc_bin.exe` which is a native executable.

In general bytecode builds are not super well supported in Coq, as Coq tends to be too slow for many uses.

**Comment by barracuda156** [comment]
@ejgallego Better slow than broken :) Thank you.

Normally with OCaml ports `world` target stands for bytecode version, so I did not think a flag is needed. Will try that.

**Comment by ejgallego** [comment]
Indeed, getting a working setup is a long way from a non working setup!

Keep me posted how the adventure goes, unfortunately you may have to tweak stuff in 8.17 again, on the good side, the change to have things working on your setup should be a one-liner in the build system.

**Comment by barracuda156** [comment]
@ejgallego Do I miss something?

It seems that the argument has no effect, I get identical configure output, and the same failure eventually:
```
--->  Configuring coq
Executing:  cd "/opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_lang_coq/coq/work/coq-8.16.1" && _build/default/tools/configure/configure.exe -prefix /opt/local -byte-only -mandir /opt/local/share/man -docdir /opt/local/share/coq/latex -browser 'open %s' -coqide no -with-doc no 
You have OCaml 4.14.0. Good!
You have OCamlfind 1.9.6. Good!
You have the Zarith library 1.12 installed. Good!
CoqIde manually disabled:
=> no CoqIDE will be built.

  Architecture                : Darwin
  Sys.os_type                 : Unix
  OCaml version               : 4.14.0
  OCaml binaries in           : /opt/local/bin/
  OCaml library in            : /opt/local/lib/ocaml
  CoqIDE                      : no
  Documentation               : None
  Web browser                 : open %s
  Coq web site                : http://coq.inria.fr/
  Bytecode VM enabled         : true
  Native Compiler enabled     : ondemand
```

**Comment by barracuda156** [comment]
I found how to disable native compiler in configure, but it did not help (same error, same spot).

Args I passed were:
```
platform powerpc {
    configure.args-append \
                    -byte-only \
                    -native-compiler no \
                    -natdynlink no
}
```

**Comment by barracuda156** [comment]
Looks like the same issue resurfaces over and again: https://github.com/coq/coq/issues/9464

**Comment by barracuda156** [comment]
@JasonGross @SkySkimmer (Tagging participants of the earlier issue with bytecode.)

**Comment by SkySkimmer** [comment]
Are you using https://github.com/coq/coq/pull/17433 ?

**Comment by SkySkimmer** [comment]
Also does `-natdynlink no` do anything since the binaries were moved to dune? IIRC it would statically link plugins in coqc but we have nothing for this in the dune files.

**Comment by barracuda156** [comment]
> Are you using #17433 ?

@SkySkimmer Ha, I guess no, since I was trying to build 8.16.1. I can try building from master then.

For `-natdynlink`, I have no idea, it was suggested by configure, I thought to try it. In any case, I got the same failure with or without all extra options. I recall OCaml had this in configure, being enabled for Darwin.

**Comment by barracuda156** [comment]
> Are you using #17433 ?

I pulled the fix over to 8.16.1 and building now. Given how far it went, that issue is solved.

**Comment by barracuda156** [comment]
@SkySkimmer @ejgallego So the build succeeded (thank you!). Trying to run tests I get this message however:
```
--->  Testing coq
Executing:  cd "/opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_lang_coq/coq/work/coq-8.16.1" && /usr/bin/make test-suite 
/usr/bin/make --warn-undefined-variable --no-builtin-rules -f Makefile.build test-suite
DUNE      sources
/usr/bin/make -C test-suite -s VERBOSE= clean
Makefile:121: cannot run complexity tests (no bogomips found)
RM        <**/*.stamp> <**/*.vo> <**/*.vio> <**/*.log> <**/*.glob>
RM        <**/*.cmx> <**/*.cmi> <**/*.o> <**/*.test>
/usr/bin/make -C test-suite -s VERBOSE= all
Makefile:121: cannot run complexity tests (no bogomips found)

Coq's standard library has not been built; please run any of: 
  - make -j NJOBS world          if you used -profile devel in configure
  - make -f Makefile.dune world  if you want to use dune autogenerated configure setup

make[2]: *** [all] Error 1
make[1]: *** [test-suite] Error 2
```
